### PR TITLE
Adds trivy scan of component images in Jenkins pipeline

### PR DESCRIPTION
**Pull request by diverdane**
Adds a trivy security scan of Marketplace app component images to project
pipeline to find any Vulnerabilities before publishing to Google Marketplace.

trivy security scans of the following Marketplace app component images
are being add to the project pipeline to find vulnerabilities before publishing to
Google Marketplace:
- deployer
- tester
- nginx
- postgres

A scan of the conjur image is skipped because this image is built and
scanned elsewhere (i.e. for the cyberark/conjur repo builds).

A .trivyignore file is being added for the following remaining
critical vulnerabilities:
- 4 critical vulnerability remaining in Google's deployer_helm container
- 2 critical vulnerabilities in postgres:9.4 container
- 2 critical vulnerabilities in nginx:1.17 container

Reference: https://github.com/aquasecurity/trivy

### CODE REVIEW

File: .trivyignore [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+CVE-2005-2541
+CVE-2017-1000082
+CVE-2019-2201
+
+
+
+
+
+
+
```
Comment (sgnn7) [inline .trivyignore]: Lots of unnecessary newlines

File: Jenkinsfile [hunk 2]
```diff
@@ -53,6 +61,38 @@ pipeline {
...
+        stage('Scan nginx image') {
+          steps {
+            scanAndReport("gcr.io/conjur-cloud-launcher-onboard/cyberark/nginx:${TAG}", "CRITICAL")
+          }
+        }
+        stage('Scan postgres image') {
+          steps {
+            scanAndReport("gcr.io/conjur-cloud-launcher-onboard/cyberark/postgres:${TAG}", "CRITICAL")
+          }
+        }
```
Comment (sgnn7) [inline Jenkinsfile]: Do we need `conjur` image here too or is that part of deployer?

Comment (diverdane) [reply Jenkinsfile]: The conjur image is being scanned when it gets built as part of the cyberark/conjur repo builds, so the recommendation (from Andy) was to skip the scan here. I put a comment in the Jenkinsfile, line 73.
I'll add a comment on this in this PR's description.

Comment (sgnn7) [reply Jenkinsfile]: Ah good catch - I totally missed that comment!

**Comment by izgeri** [comment]
can we add more context for why we're adding the critical vulns to the trivyignore? I expect we'll appreciate having written that history later. also, for adding items to trivyignore, does that require a PR approval from @andytinkham?

also, I noticed that the commit message still says
> NOTE: If we make the following image changes:
> - Downgrade to postgres:9.4 (which would be consistent with Conjur OSS)
> - Upgrace to nginx:1.17
> then the critical vulnerabilities will be reduced to 2 for both
> postgres and nginx images.

I would guess this should be updated / removed?

**Comment by andytinkham** [comment]
I agree - we absolutely want as much detail as possible in comments for why we're ignoring a CVE. It will definitely come up in the future - some customer will run Conjur through a scanner and (rightly) ask about any vulnerabilities that show up. The first place I'll look when that happens is the .trivyignore. 

I'm not opposed to reviewing every .trivyignore but don't want to become a bottleneck either. I'd definitely like the awareness though... Sounds like something worth discussing.

File: .trivyignore [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# The postgres:9.4 container is subject to the following critical
+# vulnerabilities:
+CVE-2005-2541
+CVE-2017-1000082
+
+# The nginx:1.17 container is subject to the following critical
+# vulnerabilities:
+CVE-2005-2541
+CVE-2019-2201
```
Comment (sgnn7) [inline .trivyignore:17]: I think we need to specifically have a comment line for each of these so they're easy to identify without googling

**Comment by sgnn7** [comment]
@diverdane For now though I'll just merge since it's an improvement over status quo though.

**Comment by izgeri** [comment]
@sgnn7 @diverdane I can't find on this PR the justification for why the trivyignore lines were added. can you please point me to it or add it to this PR retroactively?

for what it's worth, I have a strong preference to that being clear in the PR description or commit message **before** the PR is merged. this is important historical information that we should be tracking.

**Pull request merged**